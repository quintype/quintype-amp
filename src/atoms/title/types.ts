export interface TitleTypes {
  children: string|HTMLElement; // check if HTMLElement can be present here
}