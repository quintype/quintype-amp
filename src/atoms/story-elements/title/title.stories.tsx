import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Title } from "./title";
import { config, textStory } from "../../../__fixtures__";
import { Layout } from "../../layout";

const sampleTitleElement = {
  description: "",
  "page-url": "/story/af018d6b-2166-4344-b167-40405a148e41/element/5737bdd7-bdce-4396-9318-9ac8007f0970",
  type: "title",
  "family-id": "65d7db11-3fd5-4cf7-bcf3-a55b70b49e8a",
  title: "",
  id: "5737bdd7-bdce-4396-9318-9ac8007f0970",
  metadata: {},
  subtype: null,
  text: "This is a title"
};

storiesOf("Title", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <Title element={sampleTitleElement} />)
  .add("With custom styles", () => <Title element={sampleTitleElement} inlineStyles={{ fontStyle: "italic" }} />);
