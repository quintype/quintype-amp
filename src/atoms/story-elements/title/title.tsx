import React from "react";
import { StoryElementProps } from "../types";
import styled from "styled-components";
import { withStoryAndConfig } from "../../../context";
import get from "lodash.get";
export const StyledTitle = styled.h3`
  font-family: ${(props) => props.theme.font.family.primary};
  font-size: ${(props) => props.theme.font.size.l};
  font-weight: ${(props) => props.theme.font.weight.bold};
  line-height: ${(props) => props.theme.font.lineHeight.level6};
  margin: 0;
  text-transform: uppercase;
`;

export const TitleBase = ({ element, story, config, counter }: StoryElementProps) => {
  const titleElementRender = get(config, ["opts", "render", "storyElementRender", "titleElementRender"], null);

  return titleElementRender ? (
    titleElementRender({ story, config, element, counter })
  ) : (
    <StyledTitle>{element.text}</StyledTitle>
  );
};
/**
 * Summary is a story element.
 * The look of the Title can be changed using the render prop titleElementRender. In case titleElementRender is passed in the config, it is rendered. Otherwise a default Title is rendered.
 *
 * @param {Object} params object containing parameters passed to the render prop
 * @param {Object} params.story story object
 * @param {Object} params.config config object
 * @param {Object} params.element the story element. This is same as the story element found in the story API response
 *
 * ```javascript
 * ...
 * ampRoutes(app, {
 *  render: {
 *    storyElementRender: {
 *      titleElementRender: ({ story, config, element }) => <MyCustomTitle story={story} config={config} storyElement={element} />
 *    }
 *  }
 * })
 * ...
 * ```
 *
 * @category StoryElements
 * @module Title
 * @component
 */
export const Title = withStoryAndConfig(TitleBase);
