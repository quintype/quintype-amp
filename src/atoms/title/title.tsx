import React from "react";
import { Helmet } from "react-helmet";
import { TitleTypes } from "./types";

const Title = (props: TitleTypes) => {
  return (
    <Helmet>
      <title>{props.children}</title>
    </Helmet>
  );
};

export default Title;
