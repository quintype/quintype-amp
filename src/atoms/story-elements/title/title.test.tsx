import React from "react";
import { Title } from "./title";
import { shallow, mount } from "enzyme";

const sampleTitleElement = {
  description: "",
  "page-url": "/story/af018d6b-2166-4344-b167-40405a148e41/element/5737bdd7-bdce-4396-9318-9ac8007f0970",
  type: "title",
  "family-id": "65d7db11-3fd5-4cf7-bcf3-a55b70b49e8a",
  title: "",
  id: "5737bdd7-bdce-4396-9318-9ac8007f0970",
  metadata: {},
  subtype: null,
  text: "This is a title"
};

describe("Title", () => {
  it("should render default", () => {
    const wrapper = shallow(<Title element={sampleTitleElement} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render default", () => {
    const wrapper = mount(<Title element={sampleTitleElement} inlineStyles={{ fontStyle: "italic" }} />);
    expect(wrapper.find("h3").prop("style")).toStrictEqual({ fontStyle: "italic" });
  });
});
