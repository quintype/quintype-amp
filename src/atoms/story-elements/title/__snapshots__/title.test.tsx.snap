// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render default 1`] = `
<styled.h3>
  This is a title
</styled.h3>
`;
